### PR TITLE
TRT-569: Collecting cluster data for analysis

### PR DESCRIPTION
**Pull request by deepsm007**
https://issues.redhat.com/browse/TRT-569 

Eg: 
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/34384/rehearse-34384-pull-ci-openshift-prometheus-alertmanager-master-e2e-agnostic-cmo/1600170508496670720/artifacts/e2e-agnostic-cmo/ipi-install-install/artifacts/cluster-data.json

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/34384/rehearse-34384-pull-ci-openshift-aws-load-balancer-operator-main-e2e-aws-operator/1600226827735928832/artifacts/e2e-aws-operator/ipi-install-install/artifacts/cluster-data.json

`{"Platform": "Azure", "Topology": "HighlyAvailable", "NetworkType": "OVNKubernetes", "NetworkStack": "IPv4", "CloudRegion": "westus", "CloudZone": ["0", "1"], "ClusterVersionHistory": ["4.13.0-0.ci.test-2022-12-06-170229-ci-op-fyndfxgw-latest"]}`

### CODE REVIEW

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh [hunk 1]
```diff
@@ -385,6 +385,37 @@ if test "${ret}" -eq 0 ; then
   touch  "${SHARED_DIR}/success"
   # Save console URL in `console.url` file so that ci-chat-bot could report success
   echo "https://$(env KUBECONFIG=${dir}/auth/kubeconfig oc -n openshift-console get routes console -o=jsonpath='{.spec.host}')" > "${SHARED_DIR}/console.url"
+
+  if [ ! -f /tmp/jq ]; then
+    curl -L https://stedolan.github.io/jq/download/linux64/jq -o /tmp/jq && chmod +x /tmp/jq
+  fi
+  #save cluster data
+  touch  "${ARTIFACT_DIR}/cluster-data"
```
Comment (dgoodwin) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:393]: We will want a json formatted file for programatic parsing.

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh [hunk 2]
```diff
@@ -385,6 +385,37 @@ if test "${ret}" -eq 0 ; then
...
+  # platform
+  echo "Platform: $(env KUBECONFIG=${dir}/auth/kubeconfig oc get infrastructure/cluster -o json|/tmp/jq '.status.platform')/n" >> "${ARTIFACT_DIR}/cluster-data"
+  # topology
+  TOPOLOGY="$(env KUBECONFIG=${dir}/auth/kubeconfig oc get node -o json|/tmp/jq '.items[0]|.metadata.annotations'|grep machineconfiguration.openshift.io/controlPlaneTopology)"
+  if [[ "$TOPOLOGY" =~ .*"HighlyAvailable".*  ]]; then
+    echo "Topology: ha/n" >> "${ARTIFACT_DIR}/cluster-data"
```
Comment (dgoodwin) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:399]: What is the /n in these, and can we make it more specific in the json file?

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh (hunk 1 above)
Comment (dgoodwin) [reply ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:393]: Consider python here if the bash / jq stuff get's too unwieldy.

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh (hunk 2 above)
Comment (dgoodwin) [reply ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:399]: Are they meant to be newlines? \n

But they also may not be necessary when doing >>

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh [hunk 3]
```diff
@@ -385,6 +385,37 @@ if test "${ret}" -eq 0 ; then
...
+    echo "Network Stack: IPv6/n" >> "${ARTIFACT_DIR}/cluster-data"
+  else
+    echo "Network Stack: IPv4/n" >> "${ARTIFACT_DIR}/cluster-data"
+  fi
+  # cluster version history
+  echo "Cluster Version History: $(env KUBECONFIG=${dir}/auth/kubeconfig oc get clusterversion -o json|/tmp/jq '.items[]|.status.history'|grep version|cut -d : -f 2| sort -u)/n" >> "${ARTIFACT_DIR}/cluster-data"
+  # cloud region
+  echo "Cloud Region: $(env KUBECONFIG=${dir}/auth/kubeconfig oc get node -o json|/tmp/jq '.items[]|.metadata.labels'|grep topology.kubernetes.io/region|cut -d : -f 2| sort -u)/n" >> "${ARTIFACT_DIR}/cluster-data"
+  # cloud zone
+  echo "Cloud Zone: $(env KUBECONFIG=${dir}/auth/kubeconfig oc get node -o json|/tmp/jq '.items[]|.metadata.labels'|grep topology.kubernetes.io/zone|cut -d : -f 2| sort -u)" >> "${ARTIFACT_DIR}/cluster-data"
```
Comment (dgoodwin) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:418]: What will we get here if this doesn't exist? I assume on some platforms it won't like metal?

**Review by dgoodwin — COMMENTED** [review]
Would also love to see a sample file in the PR description once we've got it coming out in json.

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh (hunk 2 above)
Comment (deepsm007) [reply ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:399]: /n were experimental from earlier changes. They are fixed in new ones while making it a json file from a normal txt file

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh (hunk 3 above)
Comment (dgoodwin) [reply ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:418]: What will Cloud Zone be though?

**Review by droslean — APPROVED** [review]

**Pull request merged**